server/pulp/server/webservices/views/consumers.py

```diff
@@ -0,0 +1,1082 @@
+        :param consumer_id: id for the requested consumer
+        :type consumer_id: str
+
+        :return: An empty response
+        :rtype: django.http.HttpResponse
+        """
+
+        manager = factory.consumer_manager()
+        response = manager.unregister(consumer_id)
+        return generate_json_response(response)
+
+    @auth_required(authorization.UPDATE)
+    @json_body_required
+    def put(self, request, consumer_id):
+        """
+        Update a specified consumer.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: id for the requested consumer
+        :type consumer_id: str
+
+        :return: Response representing the updated consumer
+        :rtype: django.http.HttpResponse
+        """
+
+        params = request.body_as_json
+        delta = params.get('delta')
+        manager = factory.consumer_manager()
+        consumer = manager.update(consumer_id, delta)
+        add_link(consumer)
+        return generate_json_response_with_pulp_encoder(consumer)
+
+
+class ConsumerBindingsView(View):
+    """
+    View for Consumer bindings - represents the collection of
+    objects used to associate a consumer and a repo-distributor
+    association.  Users wanting to create this association will
+    create an object in this collection.  Both bind and unbind
+    is idempotent.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, consumer_id, repo_id=None):
+        """
+        Fetch all bind objects referencing the specified consumer_id. Optionally,
+        specify a repo_id to fetch all bind objects for the consumer_id to the repo_id.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: The specified consumer.
+        :type  consumer_id: str
+        :param repo_id: The repository to retrieve bindings for (optional)
+        :type  repo_id: str
+
+        :raises MissingResource: if some resource is missing
+
+        :return: Response representing the bindings
+        :rtype: django.http.HttpResponse
+        """
+
+        # Check to make sure the resources exist
+        missing_resources = {}
+        if repo_id is not None:
+            repo = factory.repo_query_manager().find_by_id(repo_id)
+            if repo is None:
+                missing_resources['repo_id'] = repo_id
+        # If get_consumer raises MissingResource we might miss reporting a bad repo_id
+        try:
+            factory.consumer_manager().get_consumer(consumer_id)
+        except MissingResource:
+            missing_resources['consumer_id'] = consumer_id
+
+        if missing_resources:
+            raise MissingResource(**missing_resources)
+
+        manager = factory.consumer_bind_manager()
+        bindings = manager.find_by_consumer(consumer_id, repo_id)
+        bindings = [serialization.binding.serialize(b) for b in bindings]
```

Comment:
I would prefer if this second value had a different name, but that might just be a style thing.